From a1ce8df56d2467d49f6a36cada662522925df5e6 Mon Sep 17 00:00:00 2001
From: Thao Luong <thao.luong.uw@rvc.renesas.com>
Date: Wed, 27 May 2015 11:41:23 +0700
Subject: [PATCH 20/25] omxvideodec: Fix a memory error due to mistakenly 
 specifying an incorrect size to g_slice_free

This patch fixes the following error when repeatedly playing a video content.

***MEMORY-ERROR***: gst-launch-1.0[979]: GSlice: assertion failed: sinfo->n_allocated > 0

refs #69206
Signed-off-by: Thao Luong <thao.luong.uw@rvc.renesas.com>
---
 omx/gstomxbufferpool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/omx/gstomxbufferpool.c b/omx/gstomxbufferpool.c
index 652eadf..385c587 100644
--- a/omx/gstomxbufferpool.c
+++ b/omx/gstomxbufferpool.c
@@ -527,7 +527,7 @@ gst_omx_buffer_pool_free_buffer (GstBufferPool * bpool, GstBuffer * buffer)
       mmngr_export_end_in_user (vdbuf_data->id_export[i]);
 #endif
 
-  g_slice_free (gboolean, omx_buf->private_data);
+  g_slice_free (GstOMXVideoDecBufferData, omx_buf->private_data);
 
   gst_mini_object_set_qdata (GST_MINI_OBJECT_CAST (buffer),
       pool->gst_omx_buffer_data_quark, NULL, NULL);
-- 
2.4.0

